### PR TITLE
Add cloud NAT

### LINKED ISSUE #35: Improve documentation of image pulling
It's not made clear that private nodes make image pulling more complicated, so this should be better documented. See #25.

### LINKED ISSUE #36: Add Cloud NAT to example project
The example project should use Cloud NAT so the private nodes can still access the internet. There should also be a boolean variable to disable it.

### PR DESCRIPTION
**Pull request by wwwil**
Close #36. Add cloud router, static address and NAT to example project with names set with input variables.

Add explanation of private node and Cloud NAT to close #35.

Signed-off-by: wwwil <wwwil.squires@gmail.com>

### CODE REVIEW

File: example/main.tf
Comment (munnerz) [inline example/main.tf]: Comment here? (maybe I just need to read the official docs, but not sure what this does..)

File: example/main.tf
Comment (munnerz) [inline example/main.tf]: So this give the NAT gateway a single IP address? Might be worth making this configurable if it's easy to do 😄

File: example/main.tf
Comment (munnerz) [inline example/main.tf]: Comment as to what this means would be great 😄

File: example/main.tf
Comment (munnerz) [inline example/main.tf]: Are there other options worth specifying here? If so, comment/making it an option would be great 😄

File: example/main.tf
Comment (munnerz) [inline example/main.tf]: Worth making this a configuration option? It seems like it could be quite an important cost-control mechanism

File: example/variables.tf
Comment (munnerz) [inline example/variables.tf:143]: Does this need to be a variable? Can you provide an example (in this description) of when you might want to override/manually specify it?

File: example/main.tf
Comment (wwwil) [reply example/main.tf]: I've now removed BGP stuff as it's not required for Cloud NAT.

File: example/main.tf
Comment (wwwil) [reply example/main.tf]: I've changed this to automatic and added a comment in the new version.

File: example/main.tf
Comment (wwwil) [reply example/main.tf]: I'm not really sure what I was doing here but it's gone now as it's not required.

File: example/main.tf
Comment (wwwil) [reply example/main.tf]: This has also been updated and is commented in the new version.

I've not made it an option as the example project is just intended to show how the module can be used and I don't want to overcomplicate it with lots of options.

File: example/main.tf
Comment (wwwil) [reply example/main.tf]: These now have variables to enable/disable and set the filter level.

File: example/variables.tf
Comment (wwwil) [reply example/variables.tf:143]: This variable is now used to explicitly enable/disable the Cloud NAT as explained in the description.

It was to set the NAT resource's name which is now fixed based on the name of the cluster. The router resource is named in the same way.

**Comment by wwwil** [comment]
I'm gonna hold off looking into test failures until this PR to improve the script is merged: https://github.com/jetstack/terraform-google-gke-cluster/pull/59

**Comment by philwinder** [comment]
Thank you for this. Saved my bacon.

**Comment by znat** [comment]
Hi there, we're really excited with this feature and look forward to it being merged 💯

File: example/main.tf
Comment (bittermandel) [inline example/main.tf]: Can we rename this to nat_enabled/enable_nat, for clarity?

**Review by bittermandel — CHANGES_REQUESTED** [review]

**Comment by wwwil** [comment]
> Hi there, we're really excited with this feature and look forward to it being merged 💯

Hey @znat! Thanks for your interest and sorry for the delay! We've had some difficulty resourcing this module, and for many of our projects have switched to using [Google's GKE module](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine).

I'm going to work on this module today to clear the backlog of PRs and release a new version, though it's likely this will be final version. We're then planning to create an example Terraform project like the example in this module, but using Google's GKE module, as well as guidance on how to migrate.

**Pull request merged**
